File: domain/src/project/properties.rs
use crate::{
    AddedBy, LastUpdatedBy, Owner,
    enums::{Status, Visibility},
};
use chrono::{DateTime, Utc};
use serde::{Deserialize, Serialize};
use sqlx::FromRow;

#[derive(Serialize, Deserialize, FromRow, Debug)]
pub struct Properties {
    pub name: String,
    pub added: DateTime<Utc>,
    #[serde(flatten)]
    pub owner: Owner,
    #[serde(flatten)]
    pub added_by: AddedBy,
    pub technologies: Vec<String>,
    pub country_code: String, // ISO 3166-1-ALPHA-2
    pub subdivisions: Vec<String>,
    pub status: Status,
    pub visibility: Visibility,
    pub crs_srid: Option<i32>,
    #[serde(flatten)]
    pub last_updated_by: LastUpdatedBy,
    pub last_updated: DateTime<Utc>,
    pub slug: String,
    pub search_area_id: Option<i32>,
    pub search_site_name: Option<String>,
}

File: react-frontend/src/domain/project/properties.ts
import type { TeamOutputDto } from "../team/outputDto";

export type Status = "ACTIVE" | "ARCHIVED" | "DELETED";
export type Visibility = "PRIVATE" | "TEAM" | "PUBLIC";



export interface ProjectProperties {
    name: string;
    added: string;
    // Owner fields (flattened)
    owner_id: number;
    owner_first_name: string;
    owner_last_name: string;
    owner_team: TeamOutputDto;
    // AddedBy fields (flattened)
    added_by_id: number;
    added_by_first_name: string;
    added_by_last_name: string;
    added_by_team: TeamOutputDto;
    // Core properties
    technologies: string[];
    country_code: string; // ISO 3166-1-ALPHA-2
    subdivisions: string[];
    status: Status;
    visibility: Visibility;
    crs_srid: number | null;
    // LastUpdatedBy fields (flattened)
    last_updated_by_id: number;
    last_updated_by_first_name: string;
    last_updated_by_last_name: string;
    last_updated_by_team: TeamOutputDto;
    last_updated: string;
    slug: string;
    search_area_id: number | null;
    search_site_name: string | null;
}